ethcontract/src/transaction/confirm.rs

```diff
@@ -189,13 +224,20 @@ impl<T: Transport> ConfirmationContext<'_, T> {
     ///
     /// This method returns the latest block number if it is known.
     async fn wait_for_blocks(&self, target_block: U64) -> Result<U64, ExecutionError> {
+        let mut cur_delay = self.params.poll_interval_min;
+
         loop {
-            delay(self.params.poll_interval).await;
-            let latest_block = self.web3.eth().block_number().await?;
+            delay(cur_delay).await;
 
+            let latest_block = self.web3.eth().block_number().await?;
             if target_block <= latest_block {
                 break Ok(latest_block);
             }
+
+            cur_delay = min(
```

Comment:
Should we reset the delay on every new block? My concern is that if we set a high number of confirmations (like 20 blocks for example) the delay might grow very high between polls.

Reply:
Given that the average block time is 14 seconds, I think polling once every 7 seconds is enough. The smaller delays will only factor when we wait for one, maybe two blocks.

Reply:
That makes sense. I'd reset it to the average delay between blocks instead of min delay.

P.S. My understanding of block times was wrong, disregard the deleted comment.

Reply:
Wait no, I've confused myself. Ok, my point is, if new blocks appear slower than `poll_interval_max`, then there's no need to reset the delay.

Reply:
Since `cur_delay` is limited by `max_delay`, I don't think we should reset it. A new block appears every five to fifteen seconds, so wait time of seven seconds is good.

Reply:
Sounds good to me. I think its a valid point that its bound by a max polling interval, and with large block confirmations you will be waiting a long time anyway.